src/abcdmicro/build_template.py

```diff
@@ -0,0 +1,428 @@
+from __future__ import annotations
+
+import tempfile
+from collections import defaultdict
+from collections.abc import Mapping, Sequence
+from pathlib import Path
+
+import ants
+import numpy as np
+from ants.core import ANTsImage, ANTsTransform
+
+from abcdmicro.resource import InMemoryVolumeResource, VolumeResource
+from abcdmicro.util import update_volume_metadata
+
+
+def average_volumes(
+    volume_list: Sequence[VolumeResource], normalize: bool = True
+) -> VolumeResource:
+    """
+    Calculates the simple arithmetic average (mean) of a list of volumes.
+    This is a simplified version of ANTs `average_images`.
+    Args:
+        volume_list: A list of VolumeResource objects to be averaged.
+                     All volumes are treated with equal weight (1/N).
+
+    Returns:
+        A VolumeResource object containing the element-wise arithmetic mean of all input volumes.
+    """
+
+    biggest = 0
+    biggestind = 0
+    for k, _v in enumerate(volume_list):
+        sz = np.prod(_v.get_array().shape)
+        if sz > biggest:
+            biggest = sz
+            biggestind = k
+
+    ref_volume = volume_list[biggestind]
```

Comment:
a bit more pythonic:

```suggestion
    ref_volume = volume_list[np.argmax([v.get_array().size for v in volume_list])]
```

also, `biggest` appears to be unused